### PR TITLE
add test to change language

### PR DESCRIPTION
**Pull request by jpita**

### CODE REVIEW

File: cypress/integration/settings-spec.js [hunk 1]
```diff
@@ -33,9 +36,28 @@ describe('settings page', () =>{
   })
 
   it('only the first element should be selected on load', () => {
-    firstElementOfTheSettingsMenuList.should('have.attr', 'nav-selected', 'true')
```
Comment (jpita) [inline cypress/integration/settings-spec.js]: for some reason this "should" was not asserting anything, I couldn't use firstElementOfTheSettingsMenuList here but on the next line it works.
if anyone can find the "why" that would be great.

Comment (hueitan) [reply cypress/integration/settings-spec.js]: it works from my side, does it work as well in CI?

Comment (jpita) [reply cypress/integration/settings-spec.js]: the test passes but the assert is never done.
you can see by running `npx cypress open` on master branch and run that test.
there is no asserts being run.
image

File: cypress/integration/settings-spec.js [hunk 2]
```diff
@@ -3,9 +3,11 @@
 
 import { SettingsPage } from "../page-objects/settings-page"
 import { SearchPage } from "../page-objects/search-page"
+import { LanguageSettingsPage } from "../page-objects/language-settings-page"
```
[the remark this answers is not in the document]
Comment (jpita) [reply cypress/integration/settings-spec.js]: are they not different?
if they're different makes sense to have 2 (both in src and tests), if they're the same (in terms of selectors) we can have just one in the tests

File: cypress/integration/settings-spec.js [hunk 3]
```diff
@@ -17,7 +19,8 @@ describe('settings page', () =>{
   })
 
   it('show all the items in the list', () => {
-    settingsPage.settingsList().should('have.text', "LanguageText sizeAbout WikipediaPrivacy policyTerms of useRate the appHelp and feedbackAbout the app")
+    settingsPage.settingsList().should('have.text', 
+    "LanguageText sizeAbout WikipediaPrivacy policyTerms of useRate the appHelp and feedbackAbout the app")
```
[the remark this answers is not in the document]
Comment (jpita) [reply cypress/integration/settings-spec.js]: I prefer your idea of getting the content from the json file but I don't know how to do that in javascript.
can you help me with that (will be fast) or should I look it up (will probably take a while)?

[the remark this answers is not in the document]
Comment (hueitan) [reply cypress/integration/settings-spec.js]: try `import * as data from '../../i18n/en.json'` in this file

File: cypress/integration/settings-spec.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (hueitan) [reply cypress/integration/settings-spec.js]: it looks the same but there are differences, [see the differences here](https://github.com/wikimedia/wikipedia-kaios/pull/58#issuecomment-568592252)

**Comment by stephanebisson** [comment]
> Do you think it's better to read the `i18n/en.json` file for all the text? Otherwise whatever translatewiki change will cause the CI failed.

`en.json` is managed by us in the code. TWN won't change it.

File: cypress/page-objects/language-settings-page.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    }
+
+    popupTitleElement(){
+        return cy.get('.language-message > .header')
+    }
+
+    headerElement(){
+        return cy.get('.header')
+    }
+}
```
Comment (stephanebisson) [inline cypress/page-objects/language-settings-page.js]: Missing a line break at the end of this file.

Comment (jpita) [reply cypress/page-objects/language-settings-page.js]: fixed

**Comment by hueitan** [comment]
true, `en.json` is mantained by us. but anyway, getting the text from `en.json` still recommended 😃

File: cypress/integration/settings-spec.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (jpita) [reply cypress/integration/settings-spec.js]: I prefer to have `LanguageSettingsPage` and `LanguageArticlePage` .
Is that ok with you?

File: cypress/integration/settings-spec.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (jpita) [reply cypress/integration/settings-spec.js]: @hueitan I added the code to use the json files for the strings, can you check now?

File: cypress/integration/settings-spec.js [hunk 5]
```diff
@@ -15,7 +37,7 @@ describe('settings page', () => {
   })
 
   it('show all the items in the list', () => {
-    settingsPage.settingsList().should('have.text', 'LanguageText sizeAbout WikipediaPrivacy policyTerms of useRate the appHelp and feedbackAbout the app')
+    settingsPage.settingsList().should('have.text', settingsMenuListEnglishText.toString().replace(/,/g, ''))
```
Comment (stephanebisson) [inline cypress/integration/settings-spec.js]: To make a string of all the elements in the array joined together, I recommend using `settingsMenuListEnglishText.join('')`

File: cypress/integration/settings-spec.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (hueitan) [reply cypress/integration/settings-spec.js]: Looks good now

File: cypress/integration/settings-spec.js [hunk 6]
```diff
@@ -31,8 +53,26 @@ describe('settings page', () => {
...
   })
+
+  it('language of the app should change', () => {
+    cy.enter()
+    languageSettingsPage.popupTitleElement().should('have.text', enJson['language-setting'])
+
+    languageSettingsPage.popupTextElement().should('have.text', languageSettingsPopupEnglishText)
+    cy.enter()
+
+    cy.get('.left').should('have.text', 'Search')
```
Comment (hueitan) [inline cypress/integration/settings-spec.js]: 'Search' here use `enJson['softkey-search']`

Comment (hueitan) [reply cypress/integration/settings-spec.js]: consider making Softkey also in the page-object, then you can `softkey.leftText()` or just `.left()`

Comment (jpita) [reply cypress/integration/settings-spec.js]: fixed

File: cypress/integration/settings-spec.js (hunk 5 above)
Comment (jpita) [reply cypress/integration/settings-spec.js]: fixed

File: cypress/integration/settings-spec.js [hunk 7]
```diff
@@ -37,7 +37,8 @@ describe('settings page', () => {
   })
 
   it('show all the items in the list', () => {
-    settingsPage.settingsList().should('have.text', settingsMenuListEnglishText.toString().replace(/,/g, ''))
+    settingsPage.settingsList().should('have.text', settingsMenuListEnglishText.join(''))
+    //settingsMenuListEnglishText.join('')
```
Comment (stephanebisson) [inline cypress/integration/settings-spec.js]: Please remove commented out code

Comment (jpita) [reply cypress/integration/settings-spec.js]: oops

Comment (jpita) [reply cypress/integration/settings-spec.js]: fixed

File: cypress/support/commands.js [hunk 8]
```diff
@@ -43,6 +43,15 @@ Cypress.Commands.add('clickCloseButton', () => {
 Cypress.Commands.add('clickSettingsButton', () => {
   cy.get('label.left').contains('Settings').click()
 })
+
 Cypress.Commands.add('navigateToHomePage', () => {
   cy.visit('http://127.0.0.1:8080')
 })
+
+Cypress.Commands.add('getLeftSoftkeyButton', () => {
+  return cy.get('.left')
```
Comment (hueitan) [inline cypress/support/commands.js]: I recommend `cy.get( '.softkey > .left' )` here to get the specific softkey left element, as `.left` may possibly exist in other component

Comment (jpita) [reply cypress/support/commands.js]: fixed and added another small improvement, does it look ok?

**Pull request merged**